### PR TITLE
Feature/add stellio flavour

### PR DESCRIPTION
**Pull request by HoucemKacem**

### CODE REVIEW

File: Flavours/stellio-flavour/common_vsilo_functionality.py [hunk 1]
```diff
@@ -0,0 +1,677 @@
...
+    global db_name
+    global v_thing_collection
+    global thing_visor_collection
+    global v_silo_id
+    global brokerspecific
+    global tenant_id
+    global is_this_vsilo_systemvsilo
+    global controllerurl
+    global adminpassword
+    global flavour_params
```
Comment (HoucemKacem) [inline Flavours/stellio-flavour/common_vsilo_functionality.py]: I added this line to fix this: 
```
print(v_silo_id + " got flavour params: " + flavour_params)
UnboundLocalError: local variable 'flavour_params' referenced before assignment
```

File: Flavours/stellio-flavour/common_vsilo_functionality.py [hunk 2]
```diff
@@ -0,0 +1,677 @@
...
+
+    # If you run this silo controller MANUALLY for DEBUG PURPOSES,
+    # then you may access the -d parameter to set a local address of the mongo DB
+    # and a local MQTT. This parameter is never visible when this silo
+    # is run via docker automatism (or similar) via the master controller.
+    parser = argparse.ArgumentParser(description='A vSilo Controller.')
+    parser.add_argument('-d', '--debug_mongo_ip', dest='debug_mongo_ip', help='run in debug mode and use the given IP address of the system MongoDB (usually 172.17.0.2)')
+    args = parser.parse_args()
+    # flavour_params must be a string
+    flavour_params = ""
```
Comment (HoucemKacem) [inline Flavours/stellio-flavour/common_vsilo_functionality.py]: I added this line to fix this:

`NameError: name 'flavour_params' is not defined`

File: Flavours/stellio-flavour/LICENSE [hunk 3]
```diff
@@ -0,0 +1,31 @@
+BSD 4-Clause License
```
Comment (bobeal) [inline Flavours/stellio-flavour/LICENSE:1]: @glx314 what license did we choose for our work?

File: Flavours/stellio-flavour/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+## How To Run
+
+### Local Docker deployment
+
+First (as admin) add a "stellio flavour" to the system, based on the Docker image available on the fed4iot dockerhub, pointing to it with argument -i. Use the VirIoT CLI as admin and run the following command  (use "f4i.py add-flavour --help" for CLI parameters).
+
+```bash  
+python3 f4i.py add-flavour -f ngsild-stellio-f -i fed4iot/ngsild-stellio-f:2.2 -d "silo with a Stellio NGSI-LD broker" -s ""
```
Comment (bobeal) [inline Flavours/stellio-flavour/README.md]: where does this 2.2 version come from?

File: Flavours/stellio-flavour/start.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+/etc/init.d/postgresql start
```
Comment (bobeal) [inline Flavours/stellio-flavour/start.sh]: Why is not PostgreSQL started like the other ones?

File: Flavours/stellio-flavour/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+WORKDIR /app
+RUN git clone https://github.com/stellio-hub/stellio-context-broker.git
+WORKDIR /app/stellio-context-broker
+
+### compile and assemble Stellio
+RUN ./gradlew build --exclude-task test
+
+### move created jars outside
+RUN mkdir ../stellio/
+RUN cp api-gateway/build/libs/api-gateway-0.5.0.jar ../stellio/
```
Comment (bobeal) [inline Flavours/stellio-flavour/Dockerfile]: Rather make it agnostic of the version (you don't want to change this Dockerfile each time we change the version of Stellio, don't you?)

File: Flavours/stellio-flavour/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+### download and configure neo4j
+RUN wget -O - https://debian.neo4j.com/neotechnology.gpg.key | apt-key add -
+RUN echo 'deb https://debian.neo4j.com stable latest' | tee -a /etc/apt/sources.list.d/neo4j.list
+RUN apt-get update
+RUN apt-get install -y neo4j
+RUN echo "dbms.default_database=stellio" >> /etc/neo4j/neo4j.conf
+RUN neo4j-admin set-initial-password neo4j_password
+
+### download and configure postgresql 10 and timescaledb 1.4.2
+RUN apt-get install -y --no-install-recommends postgresql-10
```
Comment (bobeal) [inline Flavours/stellio-flavour/Dockerfile]: We are based on PG11, not PG10

File: Flavours/stellio-flavour/README.md (hunk 4 above)
Comment (HoucemKacem) [reply Flavours/stellio-flavour/README.md]: i followed the same pattern as in the other flavours
fed4iot/broken_name-f:2.2

File: Flavours/stellio-flavour/common_vsilo_functionality.py [hunk 8]
```diff
@@ -0,0 +1,677 @@
...
+            time.sleep(3)
+
+        if not valid_silo_entry:
+            print("Error: Virtual Silo entry not found for v_silo_id:", v_silo_id)
+            exit()
+
+        try:
+            # import paramenters from DB
+            tenant_id = silo_entry["tenantID"]
+            flavourParams = silo_entry["flavourParams"]  # in this flavour, param is the silo type (Raw, Mobius, FiWare)
```
Comment (LucaPetrucci) [inline Flavours/stellio-flavour/common_vsilo_functionality.py]: I think maybe the variable flavourParams should be renamed as flavour_params, to make it work not in the debug mode.

File: Flavours/stellio-flavour/Dockerfile (hunk 6 above)
Comment (HoucemKacem) [reply Flavours/stellio-flavour/Dockerfile]: indeed

Comment (bobeal) [reply Flavours/stellio-flavour/Dockerfile]: as you have a fresh build, you can just do `RUN cp .../api-gateway-*.jar ../stellio/`

File: Flavours/stellio-flavour/common_vsilo_functionality.py (hunk 8 above)
Comment (HoucemKacem) [reply Flavours/stellio-flavour/common_vsilo_functionality.py]: changing flavourParams to flavour_params fixed the start of the silo controller.
The change was added [here](https://github.com/fed4iot/VirIoT/pull/22/commits/f5ff29718109394289bcb2ab56699c14e2e94c7d#diff-a20868b0cbf3bab921a2905d873229dcR584)

File: Flavours/stellio-flavour/README.md (hunk 4 above)
Comment (HoucemKacem) [reply Flavours/stellio-flavour/README.md]: @LucaPetrucci what version can we choose for the fed4iot/ngsild-stellio-f image ?

Comment (LucaPetrucci) [reply Flavours/stellio-flavour/README.md]: I suggest to build the docker image twice, one with tag 3.0 and the other one with tag latest.

**Review by bobeal — APPROVED** [review]

**Pull request merged**